File: src/main/java/com/example/khuasenov/fluxstarter/repository/ContactRepository.java
package com.example.khuasenov.fluxstarter.repository;

import com.example.khuasenov.fluxstarter.model.Contact;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

/**
 * @author Khusaenov on 18.07.2018
 */
@Repository
public interface ContactRepository extends MongoRepository<Contact, String> {

    public Contact getContactById(String id);
}

File: src/main/kotlin/com/example/khusaenov/fluxstarter/repository/ContactRepository.kt
package com.example.khusaenov.fluxstarter.repository

import com.example.khusaenov.fluxstarter.model.Contact
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux

/**
 * @author Khusaenov on 10.08.2018
 */
interface ContactRepository : ReactiveMongoRepository<Contact, String> {

    fun getContactByAge(age: Integer): Flux<Contact>
}